go/vt/sqlparser/ast.go

```diff
@@ -2397,7 +2397,7 @@ func ExprFromValue(value sqltypes.Value) (Expr, error) {
 		return NewIntVal(value.ToBytes()), nil
 	case value.IsFloat() || value.Type() == sqltypes.Decimal:
 		return NewFloatVal(value.ToBytes()), nil
-	case value.IsQuoted():
+	case value.IsQuoted() || value.Type() == sqltypes.Bit:
```

Comment:
I did a sanity check (based on where this function is called from); It's used to initialize the column with the default value as returned by the `describe` statement:

```
mysql> create table b(id int, bv bit(8) default 'a', primary key(id));
Query OK, 0 rows affected (0.03 sec)

mysql> describe b;
+-------+---------+------+-----+------------+-------+
| Field | Type    | Null | Key | Default    | Extra |
+-------+---------+------+-----+------------+-------+
| id    | int(11) | NO   | PRI | NULL       |       |
| bv    | bit(8)  | YES  |     | b'1100001' |       |
+-------+---------+------+-----+------------+-------+
2 rows in set (0.00 sec)
```

In other words, just parsing this as a string will yield the wrong default value.

For now, I don't think it's worth supporting this here. So, let's just return an error.